e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/record-correction-log-correct-records.spec.js

```diff
@@ -0,0 +1,151 @@
+import {
+  FEE_RECORD_STATUS,
+  FeeRecordEntityMockBuilder,
+  PENDING_RECONCILIATION,
+  UtilisationReportEntityMockBuilder,
+  RECORD_CORRECTION_REASON,
+  getFormattedMonetaryValue,
+  FeeRecordCorrectionEntityMockBuilder,
+} from '@ukef/dtfs2-common';
+import pages from '../../../../pages';
+import USERS from '../../../../../fixtures/users';
+import { today } from '../../../../../../../e2e-fixtures/dateConstants';
+import { NODE_TASKS } from '../../../../../../../e2e-fixtures';
+import { getMatchingTfmFacilitiesForFeeRecords } from '../../../../../support/utils/getMatchingTfmFacilitiesForFeeRecords';
+
+const bankId = '961';
+const reportId = 1;
+const fee = 50000;
+const correctionId1 = 1;
+const correctionId2 = 2;
+
+const firstReport = UtilisationReportEntityMockBuilder.forStatus(PENDING_RECONCILIATION).withId(reportId).withBankId(bankId).build();
+
+const firstFeeRecord = FeeRecordEntityMockBuilder.forReport(firstReport).withId(1).withStatus(FEE_RECORD_STATUS.TO_DO_AMENDED).build();
+const secondFeeRecord = FeeRecordEntityMockBuilder.forReport(firstReport)
+  .withId(2)
+  .withFacilityId('987654321')
+  .withStatus(FEE_RECORD_STATUS.TO_DO_AMENDED)
+  .withFeesPaidToUkefForThePeriod(fee)
+  .build();
+
+const correctionEntity1 = FeeRecordCorrectionEntityMockBuilder.forFeeRecordAndIsCompleted(firstFeeRecord, true)
+  .withId(correctionId1)
+  .withReasons([
+    RECORD_CORRECTION_REASON.FACILITY_ID_INCORRECT,
+    RECORD_CORRECTION_REASON.REPORTED_CURRENCY_INCORRECT,
+    RECORD_CORRECTION_REASON.REPORTED_FEE_INCORRECT,
+    RECORD_CORRECTION_REASON.UTILISATION_INCORRECT,
+    RECORD_CORRECTION_REASON.OTHER,
+  ])
+  .withPreviousValues({
+    facilityUtilisation: 600000,
+    feesPaidToUkefForThePeriod: 12345,
+    feesPaidToUkefForThePeriodCurrency: 'GBP',
+    facilityId: '123456',
+  })
+  .withCorrectedValues({
+    facilityUtilisation: 100000,
+    feesPaidToUkefForThePeriod: 1111,
+    feesPaidToUkefForThePeriodCurrency: 'JPY',
+    facilityId: '654321',
+  })
+  .build();
+
+const correctionEntity2 = FeeRecordCorrectionEntityMockBuilder.forFeeRecordAndIsCompleted(secondFeeRecord, true)
+  .withId(correctionId2)
+  .withReasons([RECORD_CORRECTION_REASON.FACILITY_ID_INCORRECT])
+  .withPreviousValues({
+    facilityUtilisation: 600000,
+    feesPaidToUkefForThePeriod: 12345,
+    feesPaidToUkefForThePeriodCurrency: 'GBP',
+    facilityId: '123456',
+  })
+  .withCorrectedValues({
+    facilityUtilisation: 100000,
+    feesPaidToUkefForThePeriod: 1111,
+    feesPaidToUkefForThePeriodCurrency: 'JPY',
+    facilityId: '654321',
+  })
```

Comment:
This isn't accurate data - a correction with one reason should only have values for that field changed, so all fields other than facility id should be null on these objects